components/StyledSwitch.js

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { typography } from 'styled-system';
+
+const CheckboxContainer = styled.div`
```

Comment:
We already have `components/InputSwitch.js` to serve this purpose, I'd recommend re-using it rather than introducing a new component.

Reply:
I assumed we wanted to deprecate this since `InputField` was marked as deprecated and it seems we're moving away from `material-ui`.

Also, InputSwitch does not support an indeterminate state, which helps convey that subscriptions maybe not be totally on or off for a particular collective.

Reply:
It should really be called `StyledSwitch` or `StyledInputSwitch` as it's not related to `InputField`, but it was tricky to do so since it doesn't implement styled-components ATM. It's tedious to implement this kind of input perfectly, making sure they work nicely on mobile, have good accessibility.

That being said, I'm not against getting rid of `material-ui` - I just wanted to prevent having 2 components to do the same thing. Feel free to open an issue and/or PR to re-add this component and replace it everywhere if you're happy with the behavior it implemented.

Reply:
@Betree I agree it is complex. I'll do it in a separate PR.